domain/model/types.go

```diff
@@ -228,6 +196,38 @@ type ModelDetailArgs struct {
 	// IsControllerModel is a boolean value that indicates if the model is the
 	// controller model.
 	IsControllerModel bool
+
+	// AgentVersion is the target version for agents running in this model.
+	AgentVersion version.Number
+}
+
+// ModelImportArgs supplies the information needed for importing a model into a
+// Juju controller.
+type ModelImportArgs struct {
+	// GlobalModelCreationArgs supplies the information needed for
+	// importing the new model into Juju.
+	GlobalModelCreationArgs
+
+	// ID represents the unique id of the model to import.
+	ID coremodel.UUID
```

Comment:
We want to move away from `ID` and start using `UUID` everywhere.

Reply:
This is as it was prior. Not in scope for this work.